presto-orc/src/main/java/com/facebook/presto/orc/OrcWriter.java

```diff
@@ -304,14 +302,43 @@ private void finishRowGroup()
         rowGroupRowCount = 0;
     }
 
-    private void writeStripe(FlushReason flushReason)
+    private void flushStripe(FlushReason flushReason)
             throws IOException
     {
-        if (stripeRowCount == 0) {
-            return;
+        List<DataOutput> outputData = new ArrayList<>();
+        long stripeStartOffset = output.longSize();
+        // add header to first stripe (this is not required but nice to have)
+        if (closedStripes.isEmpty()) {
+            outputData.add(createDataOutput(MAGIC));
+            stripeStartOffset += MAGIC.length();
+        }
+        // add stripe data
+        outputData.addAll(bufferStripeData(stripeStartOffset, flushReason));
+        // if the file is being closed, add the file footer
+        if (flushReason == CLOSED) {
+            outputData.addAll(bufferFileFooter());
         }
 
-        recordValidation(validation -> validation.addStripe(stripeRowCount));
+        // write all data
+        outputData.forEach(data -> data.writeData(output));
+
+        // open next stripe
+        columnWriters.forEach(ColumnWriter::reset);
+        dictionaryCompressionOptimizer.reset();
+        rowGroupRowCount = 0;
+        stripeRowCount = 0;
+        bufferedBytes = toIntExact(columnWriters.stream().mapToLong(ColumnWriter::getBufferedBytes).sum());
+    }
+
+    private List<DataOutput> bufferStripeData(long stripeStartOffset, FlushReason flushReason)
```

Comment:
The names `bufferStripeData` and `bufferFileFooter` kinda made some confusion with the first glance. Maybe use `create` or `get`? Essentially the data will be flushed right away instead of buffered.